src/ufs.rs

```diff
@@ -246,6 +247,88 @@ impl Ufs {
 		}
 		Ok(None)
 	}
+
+	fn iter_xattr<T>(
+		&mut self,
+		ino: &Inode,
+		mut f: impl FnMut(&ExtattrHeader, &OsStr, &[u8]) -> Option<T>,
+	) -> IoResult<Option<T>> {
+		if ino.extsize == 0 {
+			return Ok(None);
+		}
+
+		let fs = self.superblock.fsize as u64;
+		let bs = self.superblock.bsize as usize;
+		let sz = ino.extsize as usize;
+		assert!(sz < UFS_NXADDR * bs);
+
+		let mut blocks = vec![0u8; ino.extsize as usize];
+		let mut nr = 0;
+		let mut blkidx = 0;
+
+		while nr < blocks.len() {
+			let pos = ino.extb[blkidx] as u64 * fs;
+			let num = bs.min(blocks.len() - nr);
+			self.file.read_at(pos, &mut blocks[nr..(nr + num)])?;
+			blkidx += 1;
+			nr += num;
+		}
+
+		let file = Cursor::new(blocks);
+		let mut file = Decoder::new(file, self.file.config());
+		let mut name = [0u8; 64];
+		let mut data = Vec::new();
+
+		loop {
+			let begin = file.pos()?;
+			let Ok(hdr) = file.decode::<ExtattrHeader>() else {
+				break;
+			};
+			let namelen = hdr.namelen as usize;
+
+			if namelen == 0 {
+				break;
+			} else if namelen > 64 {
```

Comment:
Where does the number 64 come from?

Reply:
It comes from /usr/src/sys/ufs/ufs/extattr.h:43:
```c
#define UFS_EXTATTR_MAXEXTATTRNAME      65      /* including null */
```

Reply:
You should define that as a constant somewhere.